### PR TITLE
add snappy

### PR DESCRIPTION
**Pull request by asdfugil**

### CODE REVIEW

File: makefiles/snappy.mk [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	$(AR) -crs -- $(BUILD_STAGE)/snappy/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/libsnappy-apfs.a libsnappy.{c,m}.o; \
+	$(CC) $(CFLAGS) $(BUILD_MISC)/libiosexec/libiosexec.1.tbd -Wl,-compatibility_version,1.0.0 -Wl,-current_version,1.3.0 -shared -framework IOKit -framework CoreFoundation -framework Foundation libsnappy.{c,m}.o -o $(BUILD_STAGE)/snappy/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/libsnappy-apfs.$(LIBSNAPPY_APFS_SOVER).dylib; \
+	$(LN_S) $(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/libsnappy-apfs.$(LIBSNAPPY_APFS_SOVER).dylib $(BUILD_STAGE)/snappy/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/libsnappy-apfs.dylib; \
+	$(CC) $(CFLAGS) $(BUILD_MISC)/libiosexec/libiosexec.1.tbd snappy.o $(BUILD_STAGE)/snappy/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/libsnappy-apfs.$(LIBSNAPPY_APFS_SOVER).dylib  -framework IOKit -framework CoreFoundation -framework Foundation -o $(BUILD_STAGE)/snappy/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/bin/snappy; \
+	$(I_N_T) -change $(BUILD_STAGE)/snappy/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/libsnappy-apfs.$(LIBSNAPPY_APFS_SOVER).dylib $(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/libsnappy-apfs.$(LIBSNAPPY_APFS_SOVER).dylib $(BUILD_STAGE)/snappy/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/bin/snappy; \
```
Comment (CRKatri) [inline makefiles/snappy.mk:30]: Delete this line.

File: makefiles/snappy.mk [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	@echo "Using previously built snappy."
+else
+# library name clash with libsnappy, so snappy => snappy-apfs
+snappy: snappy-setup
+	cd $(BUILD_WORK)/snappy && \
+	$(CC) $(CFLAGS) -c libsnappy.m -o libsnappy.m.o; \
+	$(CC) $(CFLAGS) -c libsnappy.c -o libsnappy.c.o; \
+	$(CC) $(CFLAGS) -c snappy.c -o snappy.o; \
+	$(AR) -crs -- $(BUILD_STAGE)/snappy/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/libsnappy-apfs.a libsnappy.{c,m}.o; \
+	$(CC) $(CFLAGS) $(BUILD_MISC)/libiosexec/libiosexec.1.tbd -Wl,-compatibility_version,1.0.0 -Wl,-current_version,1.3.0 -shared -framework IOKit -framework CoreFoundation -framework Foundation libsnappy.{c,m}.o -o $(BUILD_STAGE)/snappy/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/libsnappy-apfs.$(LIBSNAPPY_APFS_SOVER).dylib; \
```
Comment (CRKatri) [inline makefiles/snappy.mk:27]: Use `-install_name` and `LDFLAGS`.

File: makefiles/snappy.mk [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	$(LN_S) $(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/libsnappy-apfs.$(LIBSNAPPY_APFS_SOVER).dylib $(BUILD_STAGE)/snappy/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/libsnappy-apfs.dylib; \
+	$(CC) $(CFLAGS) $(BUILD_MISC)/libiosexec/libiosexec.1.tbd snappy.o $(BUILD_STAGE)/snappy/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/libsnappy-apfs.$(LIBSNAPPY_APFS_SOVER).dylib  -framework IOKit -framework CoreFoundation -framework Foundation -o $(BUILD_STAGE)/snappy/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/bin/snappy; \
```
Comment (CRKatri) [inline makefiles/snappy.mk:29]: Use `LDFLAGS`

File: makefiles/snappy.mk [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	
+	# snappy.mk Prep libsnappy-apfs1
+	cp -af $(BUILD_STAGE)/snappy/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/libsnappy-apfs.1.dylib $(BUILD_DIST)/libsnappy-apfs1/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib
+	
+	# snappy.mk libsnappy-apfs-dev
+	cp -a $(BUILD_STAGE)/snappy/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/libsnappy-apfs.dylib $(BUILD_DIST)/libsnappy-apfs-dev/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib
+	cp -af $(BUILD_STAGE)/snappy/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/include $(BUILD_DIST)/libsnappy-apfs-dev/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)
+	
+	# snappy.mk Sign
+	$(call SIGN,snappy,snaputil.xml)
```
Comment (CRKatri) [inline makefiles/snappy.mk:52]: Sign `libsnappy-apfs1` too.

**Review by CRKatri — CHANGES_REQUESTED** [review]

**Comment by Diatrus** [comment]
I don’t know that this would serve any particular use, especially with rootless on the way.

**Comment by asdfugil** [comment]
USB mass storage devices exists.

**Comment by Diatrus** [comment]
And what does this offer for those devices snaputil doesnt?

**Comment by asdfugil** [comment]
Well it does offer more features

1. orig-fs revert shortcut
2. Show the name of the system snapshot for this boot-manifest-hash                              
3. Set the target snapshot name to be the iOS system-snapshot

**Comment by asdfugil** [comment]
All right

**Review by CRKatri — APPROVED** [review]

**Comment by CRKatri** [comment]
The PR looks good, but I agree with @Diatrus's questions about usefulness.

**Comment by asdfugil** [comment]
Also, var snapshots still works

**Comment by CRKatri** [comment]
`snaputil` allows specifying the volume...

**Comment by CRKatri** [comment]
I'm closing this because snappy is unlicensed, so we don't have permission to redistribute.

**Pull request closed without merging**